src/sentry/static/sentry/app/views/app/index.tsx

```diff
@@ -46,35 +49,38 @@ function getAlertTypeForProblem(problem) {
   }
 }
 
-class App extends React.Component {
-  static propTypes = {
-    api: PropTypes.object.isRequired,
-    routes: PropTypes.array,
-    config: PropTypes.object.isRequired,
-  };
+type Props = {
+  api: Client;
+  config: Config;
+} & RouteComponentProps<{}, {}>;
+
+type State = {
+  loading: boolean;
+  error: boolean;
+  needsUpgrade: boolean;
+  newsletterConsentPrompt: boolean;
+  user?: Config['user'];
+};
 
+class App extends React.Component<Props, State> {
   static childContextTypes = {
     location: PropTypes.object,
   };
 
-  constructor(props) {
-    super(props);
-    const user = ConfigStore.get('user');
-    this.state = {
-      loading: false,
-      error: false,
-      needsUpgrade: user && user.isSuperuser && ConfigStore.get('needsUpgrade'),
-      newsletterConsentPrompt: user && user.flags.newsletter_consent_prompt,
-    };
-  }
+  state = {
+    loading: false,
+    error: false,
+    needsUpgrade: ConfigStore.get('user')?.isSuperuser && ConfigStore.get('needsUpgrade'),
+    newsletterConsentPrompt: ConfigStore.get('user')?.flags?.newsletter_consent_prompt,
```

Comment:
This component is a bit weird because it's wrapped with `withConfig`, but only uses it to force a re-render - I did not want to refactor this in this PR.